### PR TITLE
unbreak XHPTest

### PR DESCRIPTION
**Pull request by jjergus**
This test started failing on nightly builds because `:foo:bar:baz:herp-derp::class` now apparently resolves to a correctly namespaced name (an improvement!).

### CODE REVIEW

File: tests/XHPTest.hack [hunk 1]
```diff
@@ -66,10 +66,19 @@ EOF;
...
+    ');
     expect(C\onlyx($parser->getClassNames()))->toContainSubstring(
-      /* HH_FIXME[2049] */ :foo:bar:baz:herp-derp::class,
+      :foo:bar:baz:herp-derp::class,
     );
   }
 }
+
+// This is here so that we can use a ::class literal above without FIXMEs.
+final class :foo:bar:baz:herp-derp {}
```
Comment (jjergus) [inline tests/XHPTest.hack]: This is unrelated (test behaves the same with or without it), I just added it to get rid of the FIXME.

File: tests/XHPTest.hack [hunk 2]
```diff
@@ -66,10 +66,19 @@ EOF;
   }
 
   public async function testXHPClassNamesAreCorrect(): Awaitable<void> {
-    $parser = await FileParser::fromDataAsync('<?hh class :foo:bar:baz:herp-derp {}');
-
+    // must be the same namespace as this test, otherwise the ::class literal
+    // below resolves to a differently namespaced name (except on older versions
+    // of HHVM which don't handle namespaced XHP classes correctly)
```
Comment (fredemmott) [inline tests/XHPTest.hack]: I'd much rather keep this non-namespaced, and keep the fixme or hardcode the result.

No version of the typechecker or the runtime handles XHP classes correctly - they're impossible to use.

When they actually do work, this test will fail.

Comment (jjergus) [reply tests/XHPTest.hack]: I think the purpose of this test case was to verify that the logic in definition-finder returns the same thing as ::class, so hardcoding the name would make it pointless.

There are already other test cases above that deal with hardcoded names.

I can remove the class declaration below and add back the FIXME but that won't change the test behavior.

The problem is that `:foo:bar:baz:herp-derp::class` was `xhp_...` until ~2 days ago (incorrectly), but it is `\WhateverNamespace\xhp_...` as of then (correctly).

The only other possible fix is changing the ::class literal to something like `\:foo:bar:baz:herp-derp::class` so that both the test code and the literal are in the root namespace, but I couldn't figure out how to do that without parse errors :(

Comment (fredemmott) [reply tests/XHPTest.hack]: > but it is \WhateverNamespace\xhp_... as of then (correctly).

there's not really a 'correct' value for XHP class references in namespaces presently. Perhaps a separate file with:

```
namespace {

class :facebook:defininition-finder:test:xhp-class-for-classname {
}

function facebook_definition_finder_test_xhp_class_for_classname(): classname<mixed> {
  return :facebook:definition-finder:test:xhp-class-for-classname::class;
}

}
```

... then call that non-namespaced function from the namespaced test?

> I couldn't figure out how to do that without parse errors :(

This is a good thing: every proposal to support XHP in namespaces assumes BC in namespaces doesn't need to be worried about because there is no XHP code in namespaces, as it's completely unusable in practice. Fixing the parse errors would likely kill any plan to make `:` and `\` behave similarly.

Comment (jjergus) [reply tests/XHPTest.hack]: Done. I fixed the Hack errors in the other test files so we can stop excluding them but I can revert that or add FIXMEs if you think any of the unfixed versions were preferable.

**Review by fredemmott — APPROVED** [review]
